common/buildcraft/core/BCCoreConfig.java

```diff
@@ -177,6 +179,11 @@ public static void preInit(File cfgFolder) {
         propMarkerMaxDistance.setComment("How far, in minecraft blocks, should markers (volume and path) reach?");
         none.setTo(propMarkerMaxDistance);
 
+        propPumpMaxDistance = config.get(general, "pumpMaxDistance", 64);
+        propPumpMaxDistance.setMinValue(16).setMaxValue(256);
```

Comment:
any reason the minimum is 16? if we are making this a config option anyways it might be best to set the min at 1 or 2, could be usefull for modpackmakers

Reply:
I'd assume it was copied from above, and I'm not really sure what the actual minimum and maximum values should be.

My concern is primarily confusion: if someone doesn't know that the maximum distance has been changed, then having a really small distance might be annoying...

Reply:
it could indeed be annoying if the config is set to small but i think it's more the responsability of whoever changes it to keep things sane

as for the max: 256 is 16 chunks so quite a huge distance, @Yuki-nyan you mentioned performance impacts, how much of an impact is it when you set it to things like 100, 200 and the max?

Reply:
I've only tried it in single player at 128 and 256 so far, the former was playable with around 50 pumps (with occasional freezes of a second or two) but I had allocated 64GB of RAM to Minecraft and it was using over half of that. With it set to the maximum and having the same amount of pumps, the game spends more time frozen than playable.

Reply:
i assume those pumps where all above an ocean? we might limit things at 128 to protect servers as they don't have that kind of RAM and more often then not everyone sets up his own lava pump in the nether so this would quickly put a lot of stress on servers for RAM usage

Reply:
You would assume correctly 😆 I agree lowering the limit to 128 for is probably a sensible idea for the majority of players / server admins.

Reply:
Sounds like we might want to look into optimising pumps to be able to handle larger distances properly. Although that's pretty minor, so I'll merge it in if you change the limit to 128.